src/server-extension/resolvers/ChannelsResolver/utils.ts

```diff
@@ -105,3 +106,95 @@ export function buildExtendedChannelsQuery(
 }
+
+export function buildTopSellingChannelsQuery(
+  args: TopSellingChannelsArgs,
+  info: GraphQLResolveInfo,
+  ctx: Context
+) {
+  const roundedDate = new Date().setMinutes(0, 0, 0)
+
+  const tree = getResolveTree(info)
+
+  // Extract subsquid-supported Channel sql args
+  const sqlArgs = parseSqlArguments(model, 'Channel', {
+    ...args,
+    where: args.where?.channel, // only supported WHERE part
+  })
+
+  // Extract subsquid-supported Channel fields
+  const channelSubTree = tree.fieldsByTypeName.TopSellingChannelsResult.channel
+  const channelFields = parseAnyTree(model, 'Channel', info.schema, channelSubTree)
+
+  // Generate query using subsquid's ListQuery
+  const listQuery = new ListQuery(model, ctx.openreader.dialect, 'Channel', channelFields, sqlArgs)
+  let listQuerySql = listQuery.sql
+
+  // Count NFT sells from the auctions and buy now events and add them to the query
+  listQuerySql = extendClause(listQuerySql, 'SELECT', '"top_selling_channels"."amount"')
+
+  listQuerySql = extendClause(
+    listQuerySql,
+    'FROM',
+    ` 
+           INNER JOIN (
+              WITH nftBoughtEvents AS (
+                SELECT
+                  (data#>>'{price}')::bigint AS price,
+                  data#>>'{previousNftOwner,channel}' AS previous_owner_id,
+                  "timestamp" 
+                FROM 
+                  event
+                WHERE
+                 data#>>'{isTypeOf}' = 'NftBoughtEventData'
+                 AND "timestamp" > '${new Date(
+                   roundedDate - args.periodDays * 24 * 60 * 60 * 1000
+                 ).toISOString()}'
+                 AND data#>>'{previousNftOwner,isTypeOf}' = 'NftOwnerChannel'),
+              auctionSettledEvents AS (
+                SELECT
+                  data#>>'{winningBid}' AS bid_id,
+                  data#>>'{previousNftOwner,channel}' AS previous_owner_id,
+                  bid.amount AS price,
+                  "timestamp" 
+                FROM event
+                LEFT JOIN bid ON data#>>'{winningBid}' = bid.id
+                WHERE data#>>'{isTypeOf}' IN ('EnglishAuctionSettledEventData', 'BidMadeCompletingAuctionEventData', 'OpenAuctionBidAcceptedEventData')
+                AND "timestamp" > '${new Date(
+                  roundedDate - args.periodDays * 24 * 60 * 60 * 1000
+                ).toISOString()}'
+                AND data#>>'{previousNftOwner,isTypeOf}' = 'NftOwnerChannel')
+             SELECT 
+              previous_owner_id,
+              sum(price::bigint) as amount
+             FROM (
+              SELECT 
+                price,
+                previous_owner_id FROM nftBoughtEvents
+              UNION ALL
+              SELECT 
+              price,
+              previous_owner_id FROM auctionSettledEvents
+              ) s
+              GROUP BY previous_owner_id
+              ORDER BY amount desc 
+           ) AS top_selling_channels ON top_selling_channels.previous_owner_id = channel.id`,
```

Comment:
I tried to figure out why this query is so slow even though I doesn't do any overly complex stuff.
First I rebuilt the query to just aggregate results from `events` table, without `WITH` and `UNION`, as they are not strictly required here.
It was slightly faster, but still quite slow (~2s compared to ~3s).
Then I realized that inside `2100000000000-Indexes.js` we have some indexes defined on `"event"."data"`, but they are expression indexes, ie.:
```
CREATE INDEX "events_previous_nft_owner_channel" ON "processor"."event" USING BTREE (("data"->'previousNftOwner'->>'channel'));
CREATE INDEX "events_winning_bid" ON "processor"."event" USING BTREE (("data"->>'winningBid'));
```

However, you used a dfifferent expression in your query (`#>>`), leading to those indexes being omitted which had a bad influence on the optimalization.

Here is the refactored query which addresses those issues:
```suggestion
  listQuerySql = extendClause(
    listQuerySql,
    'FROM',
    `
    INNER JOIN (
      SELECT
        (SUM((COALESCE(event.data->>'price', '0')::bigint)) + SUM(COALESCE(winning_bid.amount, 0))) AS "amount",
        "data"->'previousNftOwner'->>'channel' AS "channel_id"
      FROM "event"
      LEFT JOIN bid AS winning_bid ON "data"->>'winningBid' = winning_bid.id
      WHERE
        "event"."timestamp" > '${new Date(
          roundedDate - args.periodDays * 24 * 60 * 60 * 1000
        ).toISOString()}' AND
        "event"."data"->>'isTypeOf' IN (
          'NftBoughtEventData',
          'EnglishAuctionSettledEventData',
          'BidMadeCompletingAuctionEventData',
          'OpenAuctionBidAcceptedEventData'
        ) AND
        "data"->'previousNftOwner'->>'channel' IS NOT NULL
        GROUP BY "event"."data"->'previousNftOwner'->>'channel'
    ) AS "top_selling_channels" ON "top_selling_channels"."channel_id" = "channel"."id"
    `,
```

After applying those fixes, the query takes just ~15-20 ms to execute (tried with `periodDays: 30` and `limit: 100`)